model/metric.go

```diff
@@ -391,7 +382,7 @@ func UnescapeName(name string, scheme EscapingScheme) string {
 			var utf8Val uint
 			for j := 0; i < len(escapedName); j++ {
 				// This is too many characters for a utf8 value.
-				if j > 4 {
+				if j > 8 {
```

Comment:
Where does this come from?

Reply:
the max UTF codepoint is '\U0010FFFF', therefore the max number of characters is actually 6, not 8.

Reply:
Sorry I dived a bit and found a few constants, have a look at this https://go.dev/play/p/PoHxOOeoyQu

```
The maximum number of bytes needed to represent unicode.MaxRune is: 4
There is also utf8.UTFMAX: : 4
```

Shouldnt this remain as 4?

Reply:
Wait, it should be 7, right? The 4 bytes in the longest possible rune encode as 8 hexadecimal digits, so `j` can legally be between 0 and 7.

Reply:
Sorry for slow thinking. I think I got it now:

`utf8.UTFMAX` is about the bytes needed in UTF-8 encoded form. But what we put into the `U__` encoding is the _code point_, which is different from the encoded byte sequence. The codepoint \U0010FFFF that @ywwg mentioned would be written as `U__10FFFF`. As UTF-8 string, it would take the byte pattern `F4 8F BF BF` (the 4 bytes max length that `utf8.UTFMAX` refers to).

But this means the correct number here is 5, don't you think so?